File: Module4/Case/Furama/src/main/java/thanh/code/models/Customer.java
package thanh.code.models;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import java.sql.Date;
import java.util.Set;

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int customerId;
    @NotBlank
    private String customerName;
    private Date customerBirthday;
    private int customerGender;
    private String customerIdCard;
    private String customerPhone;
    @Email
    private String customerEmail;
    private String customerAddress;

    @ManyToOne(targetEntity = CustomerType.class)
    private CustomerType customerType;

    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
    private Set<Contract> contract;

    public Set<Contract> getContract() {
        return contract;
    }

    public void setContract(Set<Contract> contract) {
        this.contract = contract;
    }

    public int getCustomerId() {
        return customerId;
    }

    public void setCustomerId(int customerId) {
        this.customerId = customerId;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public Date getCustomerBirthday() {
        return customerBirthday;
    }

    public void setCustomerBirthday(Date customerBirthday) {
        this.customerBirthday = customerBirthday;
    }

    public int getCustomerGender() {
        return customerGender;
    }

    public void setCustomerGender(int customerGender) {
        this.customerGender = customerGender;
    }

    public String getCustomerIdCard() {
        return customerIdCard;
    }

    public void setCustomerIdCard(String customerIdCard) {
        this.customerIdCard = customerIdCard;
    }

    public String getCustomerPhone() {
        return customerPhone;
    }

    public void setCustomerPhone(String customerPhone) {
        this.customerPhone = customerPhone;
    }

    public String getCustomerEmail() {
        return customerEmail;
    }

    public void setCustomerEmail(String customerEmail) {
        this.customerEmail = customerEmail;
    }

    public String getCustomerAddress() {
        return customerAddress;
    }

    public void setCustomerAddress(String customerAddress) {
        this.customerAddress = customerAddress;
    }

    public CustomerType getCustomerType() {
        return customerType;
    }

    public void setCustomerType(CustomerType customerType) {
        this.customerType = customerType;
    }

    public Customer() {
    }
}

File: Module5/Case/FuramaNew/src/app/models/customer.ts
import {ICustomerType} from "./customer-type";
import {FormGroup, Validators} from "@angular/forms";
import {formBuilder} from "./models.module";

export class Customer implements ICustomer{
  customerAddress: string;
  customerBirthday: string;
  customerEmail: string;
  customerGender: number;
  id: number;
  customerName: string;
  customerType: ICustomerType;


  constructor(customerAddress: string, customerBirthday: string, customerEmail: string, customerGender: number, customerId: number, customerName: string, customerType: ICustomerType) {
    this.customerAddress = customerAddress;
    this.customerBirthday = customerBirthday;
    this.customerEmail = customerEmail;
    this.customerGender = customerGender;
    this.id = customerId;
    this.customerName = customerName;
    this.customerType = customerType;
  }
}

export interface ICustomer {
  id: number;
  customerName: string;
  customerBirthday: string;
  customerGender: number;
  customerEmail: string;
  customerAddress: string;

  customerType: ICustomerType;
}

export var formGroupCustomer: FormGroup = formBuilder.group(
  {
    id: [''],
    customerName: ['', [Validators.required]],
    customerBirthday: ['', [Validators.required]],
    customerGender: ['', [Validators.required]],
    customerEmail: ['',[Validators.required, Validators.email]],
    customerAddress: ['', [Validators.required]],
    customerType: ['', [Validators.required]]
  }
);
